TFNetworkLayer.py

```diff
@@ -5515,10 +5519,13 @@ def init(self, output, output_with_activation=None, target=None, layer=None):
       self.loss_norm_factor = 1.0
       if self.output.have_time_axis():
         self.output_seq_lens = output.get_sequence_lengths()
+        if self.output.is_batch_feature_major:
+          self.output = self.output.copy_with_feature_dim_axis(-1)
         time_and_batch_dims = (self.output.time_dim_axis, self.output.batch_dim_axis)
         assert time_and_batch_dims in [(0, 1), (1, 0)], "output time-batch-dim unexpected: %s" % self.output
         if output_with_activation and output_with_activation.act_func is tf.nn.softmax:
-          self.output_before_softmax_flat = flatten_or_merge(output_with_activation.x, self.output_seq_lens, time_major=output.is_time_major)
+          self.output_before_softmax_flat = flatten_or_merge(output_with_activation.x, self.output_seq_lens,
+                                                             time_major=output.is_time_major, batch_feature_major=output.is_batch_feature_major)
```

Comment:
You don't need that.
After you do `self.output.copy_with_feature_dim_axis(-1)` above, you will not have `is_batch_feature_major` anymore. So remove this.

Reply:
'self.output.copy_with_feature_dim_axis(-1)' will return the right format. But  'output_with_activation.x' will be still in the old format. So it has to be converted as well.

Reply:
Yes, but it should not be done only here. Also, `output_with_activation` should be consistent to `output`, always. Otherwise later you do not know anymore, because you just know what `self.output` is.
So, just do it where you do `copy_with_feature_dim_axis`. That makes the code much simpler (you don't need to change `_flatten_or_merge` and care about other problems in other code later).